lib/contentTypeParser.js

```diff
@@ -173,6 +173,12 @@ function rawBody (request, reply, options, parser, done) {
 }
 
 function defaultJsonParser (req, body, done) {
+  if (body === '' || body == null) {
+    const err = new Error(`Body cannot be empty when content-type is set to 'application/json'`)
+    err.statusCode = 400
+    return done(err, undefined)
+  }
```

Comment:
I'm a bit lost on why the error message in the following block does not give a good error message.